gleam: Add at 1.4.1

ianmjones opened the pull request:
**Summary**

Add the gleam package, a friendly language for building type-safe systems that scale.

Add Gleam runtime dependency rebar3 package, an Erlang build tool that makes it easy to compile and test Erlang applications and releases.

Resolves getsolus/packages#2794

**Test Plan**

* Built and installed both packages locally.
* Ran tests for Gleam Exercism exercises, and edited Gleam code to check that language server worked as expected.
* Built and ran Gleam project.

**Checklist**

- [x] Package was built and tested against unstable

EbonJaeger reviewed (changes requested):
Each package needs to be in its own commit, in the order they need to be built.

To do this from this pull request, you should be able to:

1) On your local branch, run `git reset HEAD~`. This will remove your commit, but leave all of your work in place.
2) Create a new git commit for `rebar3`
3) Create a new git commit for `gleam`
4) Run `git push --force` to update this PR branch

ianmjones commented:
Thanks for the PR review and advice @EbonJaeger, much appreciated.

I think I've worked out what you meant with regards to the `setup` step, I hadn't put 1 and 1 together when I initially hit an issue with using `%cargo_build` that `%cargo_fetch` was the solution.

I've rebuilt, installed and tested locally both packages again.

EbonJaeger commented on packages/g/gleam/package.yml:
```diff
@@ -0,0 +1,27 @@
+builddeps  :
+    rust
+rundeps    :
+    erlang
```
`erlang` is already a rundep of `rebar3`, so it doesn't need to be in the list here.

EbonJaeger replied:
Also the syntax is wrong. Lists should like like this:

```yaml
rundeps    :
    - rebar3
```

EbonJaeger commented on packages/r/rebar3/package.yml:
```diff
@@ -0,0 +1,19 @@
+    erlang
+build      : |
+    escript bootstrap
+install    : |
```
Do we want to include anything extra like shell completions or manpages? The [Fedora package](https://src.fedoraproject.org/rpms/erlang-rebar3/blob/rawhide/f/erlang-rebar3.spec#_75) installs a few other things, but the Arch AUR package only installs the shell completions.

ianmjones replied:
Thanks again for the review @EbonJaeger.

Sorry about the duff formatting of builddeps and rundeps, no idea why I did that. :facepalm: 

Good point about not needing `erlang` as separate runtime dependency for Gleam if already requiring `rebar3` with same dependency.

> Do we want to include anything extra like shell completions or manpages? The [Fedora package](https://src.fedoraproject.org/rpms/erlang-rebar3/blob/rawhide/f/erlang-rebar3.spec#_75) installs a few other things, but the Arch AUR package only installs the shell completions.

I've added the man pages and shell completions, similar to Arch AUR, although I've added fish and zsh completions as well as bash.

I had a look at the extra stuff Fedora adds, and [Nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/build-managers/rebar3/default.nix#L55), but I'm struggling to work out how they would be installed on Solus, and for the current use case, I don't think they're needed. If in the future someone with more familiarity with Erlang wants to use rebar3 from Erlang as a library, rather than our use case of as a build tool, then hopefully they'll submit an update to the package.

EbonJaeger reviewed (commented):
Sorry, one last thing. Could I get you to add [`monitoring.yml` files](https://help.getsol.us/docs/packaging/monitoring.yml/) to these packages? Thanks!

ianmjones commented:
Thanks @EbonJaeger, I've added Gleam to release-monitoring.org (so I guess I have a Fedora Project login now as my OpenID provider went away :disappointed:), added the two monitoring.yml files, and re-pushed the commits.

Back to you!

EbonJaeger reviewed (approved):
LGTM, thanks!
